sw/device/tests/flash_ctrl_idle_low_power_test.c

```diff
@@ -0,0 +1,117 @@
+                                              kDifToggleEnabled));
+
+  dif_rstmgr_reset_info_bitfield_t rstmgr_reset_info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rstmgr_reset_info));
+  CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+
+  uint32_t address = flash_ctrl_testutils_data_region_setup(
+      &flash, kRegionBasePageIndex, kFlashDataRegion, kRegionSize);
+
+  uint32_t readback_data[kNumWords];
+
+  if (rstmgr_reset_info == kDifRstmgrResetInfoPor) {
+    // Create data. Simple count for data is sufficient which will
+    // be different than the 0xFFFFFFFF that is created
+    // with an erase.
+    uint32_t data[kNumWords];
+    for (int i = 0; i < kNumWords; ++i) {
+      data[i] = i;
+    }
+
+    // Erasing the page and writing data to it followed
+    // by a read back and compare to sanity check basic operation.
+    CHECK(flash_ctrl_testutils_erase_and_write_page(
+              &flash, address, kPartitionId, data,
+              kDifFlashCtrlPartitionTypeData, kNumWords) == 0);
+    CHECK(flash_ctrl_testutils_read_page(
+              &flash, address, kPartitionId, readback_data,
+              kDifFlashCtrlPartitionTypeData, kNumWords, 0) == 0);
+    CHECK_BUFFER(data, readback_data, kNumWords);
+
+    // Setting up low power hint and starting watchdog timer followed by
+    // a flash operation (page erase) and WFI. This will create a reset
+    // at some time following the flash operation.
+    // On the second boot we can confirm that the there was no wake
+    // from low power state and also check that the erase
+    // was successful.
+    pwrmgr_testutils_enable_low_power(&pwrmgr, kDifPwrmgrWakeupRequestSourceTwo,
+                                      0);
+
+    CHECK_DIF_OK(dif_aon_timer_watchdog_stop(&aon));
+    CHECK_DIF_OK(
+        dif_aon_timer_watchdog_start(&aon, 0xffffffff, 0x40, false, false));
+
+    dif_flash_ctrl_transaction_t transaction = {
+        .byte_address = address,
+        .op = kDifFlashCtrlOpPageErase,
+        .partition_type = kDifFlashCtrlPartitionTypeData,
+        .partition_id = kPartitionId,
+        .word_count = 0x0};
+
+    CHECK_DIF_OK(dif_flash_ctrl_start(&flash, transaction));
+    wait_for_interrupt();
+
+  } else if (rstmgr_reset_info == kDifRstmgrResetInfoWatchdog) {
+    // A boot after watchdog reset but no wake from low power (expected.)
+    CHECK(flash_ctrl_testutils_read_page(
+              &flash, address, kPartitionId, readback_data,
+              kDifFlashCtrlPartitionTypeData, kNumWords, 0) == 0);
+    for (int i = 0; i < kNumWords; ++i) {
+      CHECK(readback_data[i] == 0xffffffff,
+            "Data readback following erase should be 0xFFFFFFFF");
+    }
+  } else if (rstmgr_reset_info ==
+             (kDifRstmgrResetInfoWatchdog | kDifRstmgrResetInfoLowPowerExit)) {
+    // A boot with low power exit has been detected (not expected.)
+    LOG_FATAL("Reset type shows unexpected low power exit.");
```

Comment:
`LOG_ERROR` is sufficient, given that this is indeed what the test is supposed to check.

Reply:
Redundant, but please return false from these error clauses.

Reply:
LOG_FATAL is critical / unrecoverable programming error, which is not the case here.